client.go

```diff
@@ -564,12 +570,18 @@ func (cl *Client) dialFirst(addr string, t *Torrent) (conn net.Conn, utp bool) {
 	return
 }
 
-func (cl *Client) noLongerHalfOpen(t *Torrent, addr string) {
+func (cl *Client) noLongerHalfOpen(t *Torrent, addr string) bool {
```

Comment:
same as above. self checking code. when you start to control torrent behaviour after you drop it, this code matters - torrent will behave property when something happens after you close it.